src/fuselage/CCPACSSkinSegment.cpp

```diff
@@ -0,0 +1,196 @@
+                gp_Ax1 test4(c.eFrame_eStringer.Location(), gp_Vec(c.eFrame_eStringer.Location(), point));
+                if (test4.Angle(c.eFrame_eStringer) < (45.0 * (M_PI / 180.))) {
+
+                    gp_Pnt mPnt1 = (c.sFrame_sStringer.Location().XYZ() + c.eFrame_sStringer.Location().XYZ()) / 2;
+                    gp_Pnt mPnt2 = (c.eFrame_sStringer.Location().XYZ() + c.eFrame_eStringer.Location().XYZ()) / 2;
+                    gp_Pnt mPnt3 = (c.eFrame_eStringer.Location().XYZ() + c.sFrame_eStringer.Location().XYZ()) / 2;
+                    gp_Pnt mPnt4 = (c.sFrame_eStringer.Location().XYZ() + c.sFrame_sStringer.Location().XYZ()) / 2;
+                    gp_Pnt mPnt5 = (c.sFrame_sStringer.Location().XYZ() + c.eFrame_eStringer.Location().XYZ()) / 2;
+                    gp_Pnt mPnt6(mPnt5.X(), ym, mPnt5.Z());
+
+                    gp_Ax1 ref1(mPnt1, gp_Vec(mPnt1, mPnt3));
+                    gp_Ax1 ref2(mPnt2, gp_Vec(mPnt2, mPnt4));
+                    gp_Ax1 ref3(mPnt3, gp_Vec(mPnt3, mPnt1));
+                    gp_Ax1 ref4(mPnt4, gp_Vec(mPnt4, mPnt2));
+                    gp_Ax1 ref5(mPnt6, gp_Vec(mPnt6, mPnt5));
+
+                    gp_Ax1 test5(mPnt1, gp_Vec(mPnt1, point));
+                    if (test5.Angle(ref1) < (89.0 * (M_PI / 180.))) {
+                        gp_Ax1 test6(mPnt2, gp_Vec(mPnt2, point));
+                        if (test6.Angle(ref2) < (89.0 * (M_PI / 180.))) {
+                            gp_Ax1 test7(mPnt3, gp_Vec(mPnt3, point));
+                            if (test7.Angle(ref3) < (89.0 * (M_PI / 180.))) {
+                                gp_Ax1 test8(mPnt4, gp_Vec(mPnt4, point));
+                                if (test8.Angle(ref4) < (89.0 * (M_PI / 180.))) {
+                                    double maxAngle = 0;
+
+                                    gp_Ax1 a1(mPnt6, gp_Vec(mPnt6, c.sFrame_sStringer.Location()));
+                                    if (a1.Angle(ref5) > maxAngle)
+                                        maxAngle = a1.Angle(ref5);
+                                    gp_Ax1 a2(mPnt6, gp_Vec(mPnt6, c.eFrame_sStringer.Location()));
+                                    if (a2.Angle(ref5) > maxAngle)
+                                        maxAngle = a2.Angle(ref5);
+                                    gp_Ax1 a3(mPnt6, gp_Vec(mPnt6, c.sFrame_eStringer.Location()));
+                                    if (a3.Angle(ref5) > maxAngle)
+                                        maxAngle = a3.Angle(ref5);
+                                    gp_Ax1 a4(mPnt6, gp_Vec(mPnt6, c.eFrame_eStringer.Location()));
+                                    if (a4.Angle(ref5) > maxAngle)
+                                        maxAngle = a4.Angle(ref5);
+
+                                    gp_Ax1 test9(mPnt6, gp_Vec(mPnt6, point));
+                                    if (test9.Angle(ref5) < maxAngle) {
+                                        return true;
+                                    }
```

Comment:
@bernhardmgruber Any chance, that we can simplify this brace nightmare a bit?

Reply:
yes

Reply:
:+1: